airbyte-integrations/connectors/source-shopify/integration_tests/configured_catalog.json

```diff
@@ -318,10 +318,10 @@
         "json_schema": {},
         "supported_sync_modes": ["incremental", "full_refresh"],
         "source_defined_cursor": true,
-        "default_cursor_field": ["id"]
+        "default_cursor_field": ["updated_at"]
```

Comment:
Does changing the cursor field not constitute a breaking change for existing connections? Curious as I would assume the changes made to the format of the stream's state would make it incompatible with existing state values.

Reply:
Not a big deal, because:
1) the speed of the stream is great after the migration
2) the data will be synced from the `start_date`, if there is no `STATE` value
3) I was thinking about the actual `revert` once it's needed instead of `resetting` the stream and not having the backdoor otherwise.

Personally, I don't see this as a breaking change, rather `the stream will pull the data all over again, but this data will be fulfilled with the fields the migration has`, if some Customers want to go back with the older version, they will be able to hot-switch and continue from the STATE they have, rather than spend another day or so pulling the old, slow, REST-based stream.

Reply:
Is the destination using the cursor field? I would assume that it might to set up recordsCount but I don't know the impact of changing that

Reply:
> Is the destination using the cursor field? I would assume that it might to set up recordsCount but I don't know the impact of changing that

I was testing using Postgres locally for the sake of simplicity. After the `Refresh Schema` and running the sync, after the sync for the `control` version, no issues were faced. The data is stacked on top of the existing data with no overlaps or issues. The cursor field has been propagated as expected, and the subsequent syncs are successful as well, using the new cursor field.

Reply:
> After the Refresh Schema

What happens if a user does not refresh the schema?

Reply:
If the existing stream is broken after the update, we should `Refresh schema` > `Apply changes` > sync again.

No need to `reset` the final table, it's optional here.

Reply:
Can the opt-in mechanism be scoped to a specific stream? Should we use the opt-in mechanism to tell the user this information?

Reply:
With the `breaking change`—yes, use the `Shopify-migration.md` guide and the `Smeralda.yaml` modifications accordingly. However, I don't feel there is a way to `warn` the Customer about something like this.

Reply:
I think that makes sense as the user would have to take manual action anyway if they set the thing to "Approve all changes myself"

Reply:
Ok, i'll make this as a breaking change.

Reply:
I don't think we need to. I agree with you that this is only breaking if the user has `Approve all changes myself` selected but there would anyway have to take a manual step so I don't think we need to make it a breaking change
![image](https://github.com/airbytehq/airbyte/assets/3360483/48b82cfd-bc6a-46d6-8e46-bec066483232)